app/src/main/java/com/plenart/organizeme/fragments/SignInFragment.kt

```diff
@@ -0,0 +1,100 @@
+package com.plenart.organizeme.fragments
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import com.plenart.organizeme.R
+import com.plenart.organizeme.databinding.FragmentSignInBinding
+import com.plenart.organizeme.viewModels.SignInViewModel
+
+
+class SignInFragment : Fragment() {
+    private lateinit var binding: FragmentSignInBinding
+    private val viewModel: SignInViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        binding = FragmentSignInBinding.inflate(inflater,container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initObservers()
+        initListeners()
+        getEmail()
+        getPassword()
+
+    }
+
+    private fun initListeners(){
+        binding.btnSignInSignInActivity.setOnClickListener(){
+            viewModel.signInUser();
+        }
+    }
+
+    private fun getEmail() = with(binding.etEmailSignInActivity) {
+        this.doAfterTextChanged {
```

Comment:
Just for your information: this can be done with two-way databinding (if you are using databinding)

In ViewModel something like this:  `val email = MutableLiveData<String>()`
In xml you need to have variable of view model and set that variable in fragment and then you can do: `android:text=”@={viewModel.email}”/>`